projection/proposal/proposal.go

```diff
@@ -441,40 +448,97 @@ func (projection *Proposal) HandleEvents(height int64, events []event_entity.Eve
+				return fmt.Errorf("error inserting unknown proposal into view: %v", insertProposalErr)
+			}
+
+			maybeDepositorValidatorAddress := context.maybeProposerValidatorAddress
+
+			depositorsView := NewDepositors(rdbTxHandle)
+			depositorsTotalView := NewDepositorsTotal(rdbTxHandle)
+			if insertDepositorErr := depositorsView.Insert(&view.DepositorRow{
+				ProposalId:                    *MsgSubmitUnknownProposal.MaybeProposalId,
+				DepositorAddress:              MsgSubmitUnknownProposal.ProposerAddress,
+				MaybeDepositorOperatorAddress: maybeDepositorValidatorAddress,
+				TransactionHash:               MsgSubmitUnknownProposal.TxHash(),
+				DepositAtBlockHeight:          height,
+				DepositAtBlockTime:            blockTime,
+				Amount:                        MsgSubmitUnknownProposal.InitialDeposit,
+			}); insertDepositorErr != nil {
+				return fmt.Errorf("error inserting proposer deposit record into view: %v", insertDepositorErr)
+			}
+			if updateDepositorTotalErr := depositorsTotalView.Increment(
+				*MsgSubmitUnknownProposal.MaybeProposalId, 1,
+			); updateDepositorTotalErr != nil {
+				return fmt.Errorf("error inserting proposer deposit total record into view: %v", updateDepositorTotalErr)
+			}
+
+		} else if msgSubmitProposal, ok := event.(*event_usecase.MsgSubmitProposal); ok {
+			context, err := projection.prepareNewProposalSubmissionContext(rdbTxHandle, msgSubmitProposal.Proposer)
+			if err != nil {
+				return err
+			}
+
+			depositEndTime := blockTime.Add(context.maxDepositPeriod)
+			row := view.ProposalRow{
+				ProposalId:                   *msgSubmitProposal.MaybeProposalId,
+				Title:                        "",
+				Description:                  "",
+				Type:                         msgSubmitProposal.MsgType(),
+				Status:                       view.PROPOSAL_STATUS_DEPOSIT_PERIOD,
+				ProposerAddress:              msgSubmitProposal.Proposer,
+				MaybeProposerOperatorAddress: context.maybeProposerValidatorAddress,
+				Data:                         msgSubmitProposal.Messages,
+				InitialDeposit:               msgSubmitProposal.InitialDeposit,
+				TotalDeposit:                 msgSubmitProposal.InitialDeposit,
 				TotalVote:                    big.NewInt(0),
-				TransactionHash:              MsgSubmitUnknownProposal.TxHash(),
+				TransactionHash:              msgSubmitProposal.TxHash(),
 				SubmitBlockHeight:            height,
 				SubmitTime:                   blockTime,
 				DepositEndTime:               depositEndTime,
 				MaybeVotingStartTime:         nil,
 				MaybeVotingEndTime:           nil,
 				MaybeVotingEndBlockHeight:    nil,
+				Metadata:                     msgSubmitProposal.Metadata,
 			}
 
 			if insertProposalErr := proposalsView.Insert(&row); insertProposalErr != nil {
-				return fmt.Errorf("error inserting unknown proposal into view: %v", insertProposalErr)
+				return fmt.Errorf("error inserting cancel software upgrade proposal into view: %v", insertProposalErr)
```

Comment:
Does this insert specified for cancel software upgrade proposal?

Reply:
let me update it